File: MATLAB/02631_IntrProg/week004/fermentationRate.m
function averageRate = fermentationRate(measureRate, lowerBound, upperBound)
% fermentation rates function 
% using vectorized computation
measureRate(measureRate <= lowerBound | measureRate >= upperBound) = 0;
measureRate = measureRate(measureRate ~= 0);
averageRate = sum(measureRate)/length(measureRate);
end

%% solution
% function averageRate = fermentationRate(measuredRate, lowerBound, upperBound)
% % Returns average of elements of vector measuredRate greater than
% % lowerBound and less than upperBound
% goodRates = measuredRate > lowerBound & measuredRate < upperBound;
% averageRate = mean(measuredRate(goodRates));

File: python/02631_IntrProg/week004/fermentationRate.py
import numpy as np
 
def fermentationRate(measuredRate, lowerBound, upperBound):
# Returns average of elements of vector measuredRate greater than
# lowerBound and less than upperBound
    goodRates = (measuredRate > lowerBound) & (measuredRate < upperBound)
    averageRate = np.mean(measuredRate[goodRates])
    return averageRate